master/usage/encrypt-comms.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Configuring encryption and authentication
+canonical_url: 'https://docs.projectcalico.org/master/usage/encrypt-comms'
+---
+
+## Connections from {{site.prodname}} components to etcd
+
+If you are using the etcd datastore, we recommend enabling mutual TLS authentication on
+its connections as follows.
+
+- [Configure etcd](https://coreos.com/etcd/docs/latest/op-guide/security.html) to encrypt its
+  communications with TLS and require clients to present certificates signed by the etcd certificate
+  authority.
+
+- Configure each {{site.prodname}} component to verify the etcd server's identity and to present
+  a certificate to the etcd server that is signed by the etcd certificate authority.
+  - [{{site.nodecontainer}}](../reference/node/configuration)
+  - [`calicoctl`](./calicoctl/configure/etcd)
+  - [CNI plugin](../reference/cni-plugin/configuration#etcd-location) (Kubernetes and OpenShift only)
+  - [Felix](../reference/felix/configuration#etcdv3-datastore-configuration) (on [bare metal hosts](../getting-started/bare-metal/installation/))
+  - [Typha](../reference/typa/configuration#etcdv3-datastore-configuration) (often deployed in
+    larger Kubernetes deployments)
+  - [Neutron or ML2 driver](./openstack/configuration#neutron-server-etcneutronneutronconf) (OpenStack only)
+  - [DHCP agent](https://docs.projectcalico.org/master/getting-started/openstack/) (OpenStack only)
+
+### Connections from {{site.prodname}} components to kube-apiserver (Kubernetes and OpenShift)
+
+We recommend enabling TLS on kube-apiserver, as well as the client certificate and JSON web token (JWT)
+authentication modules. This ensures that all of its communications with {{site.prodname}} components occur
+over TLS. The {{site.prodname}} components present either an X.509 certificate or a JWT to kube-apiserver
+so that kube-apiserver can verify their identities.
+
+### Connections from Felix to Typha (Kubernetes)
+
+We recommend enabling mutual TLS authentication on connections from Felix to Typha.
+To do so, you must provision Felix with a server certificate and Typha with a client
+certificate. In addition, each service will need the private key associated with their
+certificate. For additional assurance, you can also configure the following.
+
+- Generate a [SPIFFE](https://github.com/spiffe/spiffe) identifier for Felix and Typha,
+  configure `ClientURISAN` (on Typha) and `TyphaURISAN` (on Felix) to these values,
+  and generate certificates for each service with their SPIFFE ID in the
+  `URI SAN` field.
+
+- Configure `ClientCN` (on Typha) and `TyphaCN` (on Felix) to values matching the
+  `Common Name` in their client certificates.
```

Comment:
I think we should position this text as SPIFFE being the recommended option, and then Common Name as alternative if the recommended option is not possible with their chosen CA. We don't recommend both unless, as below, they are migrating from one identity scheme to another